src/ui/pages/Backup/useRecoveryPhrase.ts

```diff
@@ -0,0 +1,27 @@
+import { useQuery } from '@tanstack/react-query';
+import { walletPort } from 'src/ui/shared/channels';
+import type { BackupContext } from './types';
+
+async function getRecoveryPhrase(context: BackupContext) {
+  if (context.name === 'wallet') {
+    const mnemonic = await walletPort.request('getRecoveryPhrase', {
+      groupId: context.groupId,
+    });
+    return mnemonic.phrase;
+  } else {
+    return walletPort.request('getPendingRecoveryPhrase');
+  }
```

Comment:
Depending on the context, we either obtain a pending recovery phrase or retrieve the recovery phrase by the `groupId`